spotify.go

```diff
@@ -124,6 +171,78 @@ func decodeError(r io.Reader) error {
 // authenticate, you can use `DefaultClient`.
 type Client struct {
 	http *http.Client
+
+	autoRetry     bool
+	retryDuration time.Duration
+}
+
+func (c *Client) SetAutoRetry(flag bool) {
+	c.autoRetry = flag
+}
+
+// executeOpt executes a non-GET request. `needsStatus` describes another code
+// that can represent success. Note that in all current usages of this function,
+// we need to still allow a 200 even if we'd also like to check for a second
+// success code.
+func (c *Client) executeOpt(req *http.Request, needsStatus int, result interface{}) error {
```

Comment:
We do have many implementations of an "opt" function in the public API, but this is a private helper.  I'm not convinced the "opt" approach was the best decision anyway as it results in a pretty large API surface area, but that's what we're stuck with for now.

In any case, I don't have a strong preference either way.  I appreciate your contributions here, and if this makes the most sense to you lets keep it :+1: